integration/regression_tests.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# -----------------------------------------------------------------
+#
+# Basic binary compatibility test script for um2nc-standalone
+#
+# This script runs some basic tests to ensure binary compatibility
+# between Martix Dix's base script & changes introduced by the
+# um2nc-standalone development effort.
+#
+# Three sets of reference data are available on gadi at
+# /g/data/vk83/testing/um2nc/integration-tests. Refer to
+# /g/data/vk83/testing/um2nc/integration-tests/README.md for details.
+#
+# If running on gadi, it is recommended to
+# use the intermediate size files, by setting UM2NC_TEST_DATA to
+# /g/data/vk83/testing/um2nc/integration-tests/intermediate/<latest-version>
+#
+# Requires nccmp to be installed: https://gitlab.com/remikz/nccmp
+# On gadi, nccmp is available via:
+#
+#     module load nccmp
+#
+# -----------------------------------------------------------------
+# Terminal session setup
+# $ export UM2NC_PROJ=<um2nc-standalone git dir>
+# $ export UM2NC_TEST_DATA=<dir with test netCDFs>
+#
+# Usage:
+# cd <um2nc-standalone git dir>
+# ./integration/binary_diff.sh
+#
+# Script warns if there is a difference in netCDF data,
+# global attributes, or encodings.
+#
+# Asssumes UM2NC_TEST_DATA contains a fields file named "fields_file",
+# and two netCDFs "reference_mask.nc", "reference_nomask.nc" which
+# will be compared against.
+#
+# NB: will display some um2nc output
+
+echo "Binary equivalence/backwards compatibility diff for um2nc-standalone"
+
+if [ -z ${UM2NC_PROJ+x} ]; then
+  echo "ERROR: set UM2NC_PROJ to um2nc-standalone project dir";
+  exit
+fi
+
+if [ -z ${UM2NC_TEST_DATA+x} ]; then
+  echo "ERROR: set UM2NC_TEST_DATA to um2nc-standalone test data dir";
+  exit
+fi
+
+if [ ! -d "$UM2NC_TEST_DATA" ]; then
+  echo "ERROR: UM2NC_TEST_DATA dir does not exist";
+  exit
+fi
+
+function diff_warn {
+  # compare & warn if data, encodings, global attributes, metdatata,
+  # and history do not match.
+  echo "Comparing $1 and $2"
+  nccmp -degh "$1" "$2"
+}
+
+# input paths
+source_ff=$UM2NC_TEST_DATA/fields_file # base file to convert
+
+# Reference netCDF files
+orig_nomask_nc=$UM2NC_TEST_DATA/reference_nomask.nc
+orig_mask_nc=$UM2NC_TEST_DATA/reference_mask.nc
+
+# output paths
+out_nomask_nc=$UM2NC_PROJ/integration/nomask.nc
+out_mask_nc=$UM2NC_PROJ/integration/mask.nc
```

Comment:
This filesystem tree structure is currently not on _Gadi_ yet.
Is this how you plan to structure it?

Reply:
Currently waiting for it to be copied over to `vk83`. The test data is currently available in `/g/data/tm70/sw6175/development/um2nc-test-reference/um2nc/integration-tests`
with the structure:
```
└── 2024.11.19
    ├── full
    │   ├── aiihca.paa1jan
    │   ├── aiihca.paa1jan.nomask.orig.nc
    │   ├── aiihca.paa1jan.orig.nc
    │   ├── fields_file -> aiihca.paa1jan
    │   ├── reference_mask.nc -> aiihca.paa1jan.orig.nc
    │   └── reference_nomask.nc -> aiihca.paa1jan.nomask.orig.nc
    ├── intermediate
    │   ├── aiihca.paa1jan.intermediate
    │   ├── aiihca.paa1jan.intermediate.nomask.orig.nc
    │   ├── aiihca.paa1jan.intermediate.orig.nc
    │   ├── fields_file -> aiihca.paa1jan.intermediate
    │   ├── reference_mask.nc -> aiihca.paa1jan.intermediate.orig.nc
    │   └── reference_nomask.nc -> aiihca.paa1jan.intermediate.nomask.orig.nc
    ├── light
    │   ├── aiihca.paa1jan.subset
    │   ├── aiihca.paa1jan.subset.nomask.orig.nc
    │   ├── aiihca.paa1jan.subset.orig.nc
    │   ├── fields_file -> aiihca.paa1jan.subset
    │   ├── reference_mask.nc -> aiihca.paa1jan.subset.orig.nc
    │   └── reference_nomask.nc -> aiihca.paa1jan.subset.nomask.orig.nc
    └── README.md
```
In script, we'd want to set `UM2NC_TEST_DATA` to point one of the above directories based on the command line argument. 

We'll need to point to the version too, which perhaps could be a second command line argument that defaults to the latest one?

For the temporary output files `out_nomask_nc` and `out_mask_nc`, it's currently placing them below the `$UM2NC_PROJ` directory, which has been fine when running the scripts locally and manually. When removing this variable I'll add a required command line argument which specifies a location to create the temporary output.

Reply:
This looks fine, but instead of `2024.11.19` I think the "version" folder should be named according to `um2nc`'s "to-be-released" version (e.g. `1.0` or `1.0.0` - depending what format we end up choosing - for this first release).

Also, since the script is supposed to be run within a Python environment with installed the version of `um2nc` we want to test, we might be able to link the that version directly (this is not super straightforward because the version is inferred with `versioneer` that doesn't use the actual "to-be-released" version for a "dev" package, which is going to be the state of the package that will be used for the tests. Anyway, we can think of a good way to set this).